cmd/build-webclient/build-webclient.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// The build-webclient tool generates the static resources needed for the
+// web client (code at client/web).
+//
+// # Running
+//
+// Meant to be invoked from the tailscale/web-client-prebuilt repo when
+// updating the production built web client assets. To run it manually,
+// you can use `./tool/go run ./misc/build-webclient`
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"tailscale.com/util/precompress"
+)
+
+func main() {
+	flag.Parse()
+
+	// The toolDir flag is relative to the current working directory,
+	// so we need to resolve it to an absolute path.
+	toolDir, err := filepath.Abs("./tool")
+	if err != nil {
+		log.Fatalf("Cannot resolve tool-dir: %v", err)
+	}
+
+	if err := build(toolDir, "client/web"); err != nil {
+		log.Fatalf("%v", err)
+	}
+}
+
+func build(toolDir, panelDir string) error {
+	if err := os.Chdir(panelDir); err != nil {
+		return fmt.Errorf("Cannot change cwd: %w", err)
+	}
+
+	if err := yarn(toolDir); err != nil {
+		return fmt.Errorf("install failed: %w", err)
+	}
+
+	if err := yarn(toolDir, "lint"); err != nil {
+		return fmt.Errorf("lint failed: %w", err)
+	}
+
+	if err := yarn(toolDir, "build"); err != nil {
+		return fmt.Errorf("build failed: %w", err)
+	}
+
+	var compressedFiles []string
+	if err := precompress.PrecompressDir("build", precompress.Options{
+		ProgressFn: func(path string) {
+			log.Printf("Pre-compressing %v\n", path)
+			compressedFiles = append(compressedFiles, path)
+		},
+	}); err != nil {
+		return fmt.Errorf("Cannot precompress: %w", err)
+	}
+
+	// Cleanup pre-compressed files.
+	for _, f := range compressedFiles {
+		if err := os.Remove(f); err != nil {
+			log.Printf("Failed to cleanup %q: %v", f, err)
+		}
+		// Removing intermediate ".gz" version,
+		// we use ".br" asset, which comes out smaller.
+		if err := os.Remove(f + ".gz"); err != nil {
+			log.Printf("Failed to cleanup %q: %v", f+".gz", err)
+		}
```

Comment:
Using the `.br` version worked fine for me running things locally, but not sure if removing the `.gz` version will cause any issues for other platforms. Looks like when serving, the `precompress` util checks `tsweb.AcceptsEncoding(r, "br")`

Reply:
IE11 doesn't support brotli: https://caniuse.com/brotli.  If we don't have at least the gz or some other versions, then this will break the web client entirely on unsupported browsers.

I can't remember what our browser support goal was.  How much additional savings does brotli get us over gzip?

Reply:
Its minimal - a couple kb. The other thing we could do is include both in the output (combined still smaller than the original files, and shouldn't grow as quickly as the original files)

Reply:
Updated to include both for now

Reply:
Went back and forth here in the most recent update. Landed on _only_ using gzip